File: app/src/main/java/com/walkingtale/Analytics.java
package com.walkingtale;

import android.content.Context;

import com.google.firebase.analytics.FirebaseAnalytics;

public class Analytics {
    private static Analytics instance;
    private static FirebaseAnalytics mFirebaseAnalytics;

    public static void init(Context context) {
        instance = new Analytics();
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(context);
    }

    public static FirebaseAnalytics getInstance() {
        return mFirebaseAnalytics;
    }
}

File: app/src/main/java/com/talkingwhale/Analytics.kt
package com.talkingwhale

import android.content.Context
import android.os.Bundle

import com.google.firebase.analytics.FirebaseAnalytics

object Analytics {
    private lateinit var mFirebaseAnalytics: FirebaseAnalytics

    fun init(context: Context) {
        mFirebaseAnalytics = FirebaseAnalytics.getInstance(context)
    }

    fun logEvent(eventType: EventType, TAG: String, message: String) {
        val bundle = Bundle()
        bundle.putString(TAG, message)
        mFirebaseAnalytics.logEvent(eventType.toString(), bundle)
    }

    fun logEvent(eventType: EventType, TAG: String) {
        logEvent(eventType, TAG, "")
    }

    enum class EventType {
        CreatedUser,
        UserLogin,
        PlayedStory,
        CreatedStory
    }
}
